lib/ar_lazy_preload/active_record/relation.rb

```diff
@@ -1,12 +1,45 @@
 # frozen_string_literal: true
 
 require "ar_lazy_preload/context"
+require "ar_lazy_preload/contexts/temporary_preload_config"
 
 module ArLazyPreload
   # ActiveRecord::Relation patch with lazy preloading support
   module Relation
     attr_writer :preloads_associations_lazily
 
+    def preload_associations(records) # rubocop:disable Metrics/MethodLength :nodoc:
```

Comment:
> This PR is a draft and not ready to be merged because won't work for rails 4.x, 5.x

We could wrap this code to the condition checking that Rails is >= 6.0, and just keep the existing behavior for older versions (and add a note to the docs).

Reply:
sounds good to me